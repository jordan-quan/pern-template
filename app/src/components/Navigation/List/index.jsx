import React from 'react'
import { NavLink } from 'react-router-dom'
import { Col, Row } from 'antd'

import * as routes from 'constants/routes'

/**
 * Basic link list
 */
const List = () => (
  <Row>
    <Col>
      <NavLink to={routes.DASHBOARD} exact activeClassName="selected">
        Dashboard
      </NavLink>

      <NavLink to={routes.ROUTES} activeClassName="selected">
        Routes
      </NavLink>

      <NavLink to={routes.FIFA} activeClassName="selected">
        FIFA
      </NavLink>

      <NavLink to={routes.UTILS} activeClassName="selected">
        Utils
      </NavLink>
    </Col>
  </Row>
)

export default List
