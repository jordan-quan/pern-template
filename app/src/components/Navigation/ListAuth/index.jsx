import React from 'react'
import { any } from 'prop-types'
import { NavLink } from 'react-router-dom'
import { Row, Col } from 'antd'

import * as routes from 'constants/routes'

/**
 * Authenticated, admin level list
 *
 * Currently unused
 */
const ListAuth = ({ session }) => (
  <Row>
    <Col>
      <NavLink to={routes.ACCOUNT} activeClassName="selected">
        Account {session.me?.username}
      </NavLink>

      {session && session.me?.role === 'ADMIN' && (
        <NavLink to={routes.ADMIN} activeClassName="selected">
          Admin
        </NavLink>
      )}
    </Col>
  </Row>
)

ListAuth.propTypes = {
  session: any
}
ListAuth.defaultProps = {
  session: null
}

export default ListAuth
