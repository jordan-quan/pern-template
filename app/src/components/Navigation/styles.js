import styled from 'styled-components'

import { SIDEBAR_WIDTH } from 'constants/.'

const colors = {
  gray: 'gray',
  purple: 'purple',
  text: 'darkgray'
}

export const Sidebar = styled.div`
  width: ${SIDEBAR_WIDTH};
  height: 100%;
  background: ${colors.gray};
  border-right: 1px solid rgba(216, 216, 216);
  position: fixed;
  overflow: hidden;

  .selected {
    border-color: ${colors.purple};
    color: ${colors.purple};
    font-weight: 500;
  }

  a {
    line-height: 35px;
    border-left: solid 5px;
    border-color: transparent;
    text-decoration: none;
    color: ${colors.text};
    padding: 0px 25px;
  }

  a:visited {
    color: ${colors.text};
  }

  .selected:visited {
    color: ${colors.purple};
  }
`

export const SideHeader = styled.div`
  padding: 40px 35px 10px 15px;
  svg {
    width: 100%;
  }
`
