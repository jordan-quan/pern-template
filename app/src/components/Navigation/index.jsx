import React from 'react'
import { any } from 'prop-types'
import { Col, Row } from 'antd'

import ListAuth from './ListAuth'
import List from './List'
import { ReactComponent as TelusLogo } from 'assets/telusLogo.svg'
import * as styles from './styles'

const Navigation = ({ session }) => (
  <styles.Sidebar>
    <Row>
      <Col>
        <styles.SideHeader>
          <TelusLogo />
        </styles.SideHeader>
        <List />
        {session?.me && <ListAuth session={session} />}
      </Col>
    </Row>
  </styles.Sidebar>
)

Navigation.propTypes = {
  session: any
}
Navigation.defaultProps = {
  session: null
}

export default Navigation
