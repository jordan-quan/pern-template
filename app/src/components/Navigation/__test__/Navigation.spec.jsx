import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { shape } from 'prop-types'
import { BrowserRouter } from 'react-router-dom'
import { MockedProvider } from '@apollo/client/testing'

import Navigation from '../index'

const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {}
  }
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) }
})

function mountWrap(node) {
  return mount(node, createContext())
}

let props, wrapper
const createProps = () => ({ session: { me: null } })
const createAuthProps = () => ({ session: { me: shape({}) } })

beforeEach(() => {
  props = createProps()
  wrapper = mountWrap(<Navigation {...props} />)
})

describe('<Navigation/>', () => {
  it('should render Navigation correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Navigation {...props} />
        </BrowserRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render Telus Logo tag', () => {
    expect(wrapper.find('svg').exists()).toBeTruthy()
  })

  it('should render List component', () => {
    expect(wrapper.find('List').exists()).toBeTruthy()
  })

  it('should render ListAuth component', () => {
    props = createAuthProps()
    wrapper = mountWrap(
      <MockedProvider>
        <Navigation {...props} />
      </MockedProvider>
    )
    expect(wrapper.find('ListAuth').exists()).toBeTruthy()
  })
})
