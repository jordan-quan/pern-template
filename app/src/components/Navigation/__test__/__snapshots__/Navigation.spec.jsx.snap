// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation/> should render Navigation correctly 1`] = `
<div
  className="sc-AxjAm hlLNoe"
>
  <div
    className="TDS_Box-modules__betweenBottomMargin-4___2tJ2n TDS_Box-modules__stack___33m4D"
  >
    <div
      className="sc-AxirZ hRPEQs"
    >
      <svg>
        telusLogo.svg
      </svg>
    </div>
    <div
      className="TDS_Box-modules__betweenBottomMargin-3___1jBrz TDS_Box-modules__stack___33m4D"
    >
      <a
        aria-current="page"
        className="selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Dashboard
      </a>
      <a
        aria-current={null}
        href="/routes"
        onClick={[Function]}
      >
        Routes
      </a>
      <a
        aria-current={null}
        href="/proactive"
        onClick={[Function]}
      >
        Proactive
      </a>
      <a
        aria-current={null}
        href="/fifa"
        onClick={[Function]}
      >
        FIFA
      </a>
      <a
        aria-current={null}
        href="/utils"
        onClick={[Function]}
      >
        Utils
      </a>
    </div>
  </div>
</div>
`;
